rules_conditions.md

```diff
@@ -0,0 +1,224 @@
+
+- `Field` (or `Fields`)
+- `Operator`
+- `Value`
+- `Datatype`
+
+The `Operator` is a required parameter, and controls which of the condition's other elements are required and which are optional.
+
+## `Field`
+
+The `Field` parameter points to a specific named element in the trace data.
+If a `Field` is named within a span, then it `exists`.
+A specific `Field` may or may not exist on any specific span in a trace.
+It might not even exist within a trace at all.
+
+When a `Field` is absent in all spans within a trace, the associated rule does not apply to that trace.
+
+A `Field` is always a single string.
+A `Field` is always matched by exact comparison.
+No transformations for case or punctuation are performed.
+
+### Example use of `Field`
+
+```yaml
+Conditions:
+    Field: http.route
+    Operator: =
+    Value: /health-check
+```
+### Leveraging Special Refinery Telemetry in Root Spans
+
+Some Refinery configuration options introduce special fields that are added to telemetry.
+
+For example, when `AddCountsToRoot` is enabled, `meta.span_count` is added to all root spans, and allows for the creation of rule conditions based on span counts.
+In this `meta.span_count` example, the Refinery rule applies to traces with more than 300 spans.
+
+```yaml
+Conditions:
+    Field: "meta.span_count"
+    Operator: ">"
+    Value: 300
+    Datatype: int
+```
+
+For details about all supported special fields, check out our [Refinery Telemetry documentation](https://docs.honeycomb.io/manage-data-volume/refinery/configuration/#refinery-telemetry).
+
+### Virtual Fields
+
+To handle specific scenarios when rules are evaluated before the arrival of root spans, Refinery introduces the concept of virtual fields. These fields provide metadata about traces that have timed out while waiting for their root span.
+
+This example shows a rule that drops traces containing more than 1000 spans by using the virtual field `?.NUM_DESCENDANTS`.
+
+```yaml
+Rules:
+    - Name: Drop any big traces
+      Drop: true
+      Conditions:
+        Field: "?.NUM_DESCENDANTS"
+        Operator: ">"
+        Value: 1000
+        Datatype: int
+
+```
+
+#### Supported Virtual Fields
+
+All virtual fields are prefixed with `?.` to distinguish them from normal fields.
+Currently only one virtual field is supported.
+
+- `?.NUM_DESCENDANTS`: the current number of child elements contained within a trace.
+
+## `Fields`
+
+The `Fields` parameter allows a single rule to apply to the first match among multiple field names.
+It is typically used when telemetry field names are being changed.
+It is exactly equivalent to `Field`, except that it must be expressed as an array of strings instead of a single value.
+The array defines a sequences of `Field` names that are checked in order for each span being considered.
+The first field that `exists` on any given span is used for the condition.
+
+### Example use of `Fields`
```

Comment:
Can I use `Fields` if I only have one?

If so, should we simply promote the use of `Fields` for all cases?

Reply:
Because of YAML and Go, you have to express Fields as an array, while Field can only be a single value. That means that all existing refinery rules would have to change. I would have preferred to have just one way to do it, but I couldn't do that and keep backwards compatibility.

Reply:
Ah shucks